### PR TITLE
Makefile

### PR DESCRIPTION
**Pull request by fcrisciani**
Bump the golang to 1.8.3
Added new code quality checks
Fixed issue highlighted

### CODE REVIEW

**Comment by fcrisciani** [comment]
@mavenugo @sanimej can you take a look to this one? 
Especially on the fix of the code issues to see that I capture the correct behavior

File: drivers/overlay/ov_network.go [hunk 1]
```diff
@@ -781,15 +781,10 @@ func (n *network) Value() []byte {
 		netJSON = append(netJSON, sj)
 	}
 
-	b, err := json.Marshal(netJSON)
-	if err != nil {
-		return []byte{}
-	}
-
```
Comment (mavenugo) [inline drivers/overlay/ov_network.go]: How is this change related to the PR ?

File: drivers/solaris/overlay/ov_network.go [hunk 2]
```diff
@@ -570,15 +570,10 @@ func (n *network) Value() []byte {
 		netJSON = append(netJSON, sj)
 	}
 
-	b, err := json.Marshal(netJSON)
-	if err != nil {
-		return []byte{}
-	}
-
```
Comment (mavenugo) [inline drivers/solaris/overlay/ov_network.go]: Same here

File: drivers/overlay/ov_network.go (hunk 1 above)
Comment (fcrisciani) [inline drivers/overlay/ov_network.go]: Is the ineffassign tool, b is defined and set but not used so it mark is as an error.
But, I remember that in the past @aboch was mentioning it as a way to validate the object, possible?

File: drivers/solaris/overlay/ov_network.go (hunk 2 above)
Comment (fcrisciani) [inline drivers/solaris/overlay/ov_network.go]: same here

File: drivers/overlay/ov_network.go (hunk 1 above)
Comment (fcrisciani) [inline drivers/overlay/ov_network.go]: Just checked that is simply unused code. Look at line: https://github.com/docker/libnetwork/blob/master/drivers/overlay/ov_network.go#L883 b is redefined without being used, and here https://github.com/docker/libnetwork/blob/master/drivers/overlay/ov_network.go#L881 netJSON is assigned in the non Marshalled version

Comment (mavenugo) [inline drivers/overlay/ov_network.go]: Fair enough

File: drivers/solaris/overlay/ov_network.go (hunk 2 above)
Comment (mavenugo) [inline drivers/solaris/overlay/ov_network.go]: same here . ;-)

**Comment by mavenugo** [comment]
LGTM (again)
